src/collective/ckeditor/browser/ckeditorview.py

```diff
@@ -184,35 +186,25 @@ def getCK_contentsCss(self):
         """
         return list of style sheets applied to ckeditor area
         the list is returned as a javascript string
-        by default portal_css mixin + plone_ckeditor_area.css
+        by default all bundled css items from resource registry + plone_ckeditor_area.css
         TODO : improve it with a control panel
         """
-        context = aq_inner(self.context)
-        portal = self.portal
-        portal_url = self.portal_url
-        portal_css = getToolByName(portal, 'portal_css')
-        css_jsList = "["
-        current_skin = context.getCurrentSkinName()
-        skinname = url_quote(current_skin)
-        css_res = portal_css.getEvaluatedResources(context)
-        for css in css_res:
-            media = css.getMedia()
-            rel = css.getRel()
-            if media not in ('print', 'projection') and rel == 'stylesheet':
-                cssPloneId = css.getId()
-                if ABSOLUTE_URL.match(cssPloneId):
-                    css_jsList += "'%s', " % cssPloneId
-                else:
-                    cssPlone = '%s/portal_css/%s/%s' % (portal_url,
-                                                        skinname,
-                                                        cssPloneId)
-                    css_jsList += "'%s', " % cssPlone
-
-        baseres = '++resource++ckeditor_for_plone'
-        css_jsList += "'%s/%s/ckeditor_plone_area.css']" % (portal_url,
-                                                            baseres)
-
-        return css_jsList
+        css_jsList = []
+        # retrieve plone.htmlhead.links viewlet manager
+        view = View(self.portal, self.request)
+        manager_name = 'plone.htmlhead.links'
+        # viewlet managers are found by Multi-Adapter lookup
+        adapter = queryMultiAdapter((self.portal, self.request, view), IViewletManager, manager_name, default=None)
+        if adapter:
+            # TODO: can we lookup the viewlet directly? (what if it was transferred to another manager?)
```

Comment:
I am not completely persuaded yet. I thought, the viewlet-manager-way was faster because the viewlet already got initialised (though i am not a 100% sure it was). With the other way (get registration, instantiate viewlet, use viewlet), the viewlet is newly initialised and loads all the stuff from the registration, that has already been loaded once. Should I still switch to that way?

Reply:
Ah. I didn't realise we are in a completely separate request (for the iframe content). So the viewlet has not been instantiated yet. I'll switch to the method shown in the documentation.